vmware-base/src/com/cloud/hypervisor/vmware/mo/VmdkFileDescriptor.java

```diff
@@ -27,55 +27,53 @@
 
 import org.apache.log4j.Logger;
 
+import com.cloud.utils.StringUtils;
+import com.google.common.base.Preconditions;
+
 public class VmdkFileDescriptor {
     private static final Logger s_logger = Logger.getLogger(VmdkFileDescriptor.class);
     private static final String VMDK_PROPERTY_CREATE_TYPE = "createType";
     private static final String VMDK_CREATE_TYPE_VMFSSPARSE = "vmfsSparse";
     private static final String VMDK_PROPERTY_ADAPTER_TYPE = "ddb.adapterType";
 
-    private Properties _properties = new Properties();
+    private final Properties _properties = new Properties();
     private String _baseFileName;
 
     public VmdkFileDescriptor() {
     }
 
     public void parse(byte[] vmdkFileContent) throws IOException {
-        BufferedReader in = null;
-        try {
-            in = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(vmdkFileContent),"UTF-8"));
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(vmdkFileContent),"UTF-8"));) {
             String line;
             while ((line = in.readLine()) != null) {
                 // ignore empty and comment lines
                 line = line.trim();
-                if (line.isEmpty())
-                    continue;
-                if (line.charAt(0) == '#')
+                if (line.isEmpty() | line.charAt(0) == '#') {
```

Comment:
Yes it is; it is a binary operator. The "|" is a bitwise OR operation.

@DaanHoogland are you sure you wanted to use a "|"?
I have just glanced at the code, but it seems that you should have used the "||" (the or operator)